app/src/js/components/Rules/rule.js

```diff
@@ -44,190 +46,166 @@ const breadcrumbConfig = [
-      postActionModal: true,
-      postActionText: enableConfirm(ruleName),
-      success: this.reload
-    }, {
-      text: 'Disable',
-      action: this.disable,
-      disabled: data.type === 'onetime',
-      status: disabledStatus,
-      confirmAction: true,
-      postActionModal: true,
-      confirmText: disableText(ruleName),
-      postActionText: disableConfirm(ruleName),
-      success: this.reload
-    }, {
-      text: 'Delete',
-      action: this.delete,
-      status: deleteStatus,
-      success: this.navigateBack,
-      confirmAction: true,
-      confirmText: deleteText(ruleName)
-    }, {
-      text: 'Rerun',
-      action: this.rerun,
-      status: rerunStatus,
-      success: this.reload,
-      confirmAction: true,
-      confirmText: rerunText(ruleName)
-    }];
-
-    const errors = this.errors();
-    return (
-      <div className='page__component'>
-        <section className='page__section page__section__controls'>
-          <div className="options--top">
-            <Breadcrumbs config={breadcrumbConfig} />
-          </div>
-        </section>
-        <section className='page__section page__section__header-wrapper'>
-          <div className='page__section__header'>
-            <h1 className='heading--large heading--shared-content with-description'>Rule: {ruleName}</h1>
-            <DropdownAsync config={dropdownConfig}/>
-
-            <Link
-              className='button button--copy button--small button--green form-group__element--right'
-              to={(location) => ({
-                pathname: '/rules/add',
-                search: getPersistentQueryParams(location),
-                state: {
-                  name: ruleName
-                }
-              })}>Copy Rule</Link>
-            <Link
-              className='button button--edit button--small button--green form-group__element--right'
-              to={(location) => ({ pathname: `/rules/edit/${ruleName}`, search: getPersistentQueryParams(location) })}>Edit Rule</Link>
-            {lastUpdated(data.timestamp || data.updatedAt)}
-          </div>
-        </section>
-        <section className='page__section'>
-          {errors.length > 0 && <ErrorReport report={errors} />}
-          <div className='heading__wrapper--border'>
-            <h2 className='heading--medium with-description'>Rule Overview</h2>
-          </div>
-          <div className="rule__state">
-            {data.state && (
-              <dl className='status--process'>
-                <dt>State:</dt>
-                <dd className={`status__badge status__badge--${data.state.toLowerCase()}`}>{displayCase(data.state)}</dd>
-              </dl>
-            )}
-          </div>
-          <div className="rule__content">
-            <Metadata data={data} accessors={metaAccessors} />
-          </div>
-        </section>
-      </div>
-    );
+  if (!record || (record.inflight && !record.data)) {
+    return <Loading />;
   }
-}
+
```

Comment:
Another big block that looks like a move. Is there anything changed in here?

Reply:
I had to refactor from class to functional component to implement React Router. No functionality changes just changed to standard. They have to be in a particular call order for React. The useEffect has to go first and all...then useMemo...